### ISSUE TITLE
IE Crusher Taking A Long Time To Crush

### ISSUE THREAD

**Issue by Pikachyuu** [NillerMedDild/Enigmatica2Expert#1858]
*Didn't mean to close it.

**Modpack version**
1.82a

**Issue**
I know that this issue has been brought up before but I have some information to shed on the subject matter. So after coming across the bug myself while trying to turn coal into coal dust I decided to do a bit of experimenting. My first thought was to just start throwing a bunch of different items in and see if any of them would crush. The results of that is that there were certain items that would crush at the appropriate speed and others that wouldn't. After looking at the commonalities between the items I came to the conclusion that Items with a modded primary output take virtually no RF (I would say around 20RF per second) resulting in the long crushing time on the contrary items with a vanilla primary output take the appropriate amount of RF in the appropriate amount of time. There is an exception though. Crushed primary outputs from 'Immersive Engineering' itself seem to work perfectly fine. I should also add that the secondary output does not seem to impede the process at all since it seems to work with both vanilla and non-vanilla items. The input material does not seem to matter as I have tested it with both vanilla and non-vanilla items. I would assume the cause of this would have something to do with the link between the localized outputs and modded ones. Something which is causing the RF input amount into the machine to be much lower when dealing with non-localized outputs rather then with localized outputs. I would assume it's some sort of mod interference but there are other possibilities as well. I will continue with further testing but for now this is as much as I have figured out.  If I have any other updates I will post them here.

Update 1: After tossing in some uranium I expected the output (which is uranium grit) to be almost instant since it is apart 'Immersive Engineering' itself but I ended up getting different results. It has the same issue as the other modded items.

Update 2: Some more research and it seems that this persons observations coincide with mine. This was the one comment in the issue thread that Niller put up.
Source: https://github.com/BluSunrize/ImmersiveEngineering/issues/4435

(Fixed grammatical errors)

"By looking at the recipe system I see two things:

1st) The crusher does not seem to use the time modifier at all only an energy modifier.
2nd) There seems to be something wrong in the calculation.

CrusherRecipe r = new CrusherRecipe(out, denseOre, (int)(recipe.getTotalProcessEnergy()/CrusherRecipe.energyModifier)*2);

Isn't the modifier to be used as a product instead of an division?!"

I don't know anything about the last statement in the quote but it the problem does seem to be the Energy input. My question is why it works with vanilla recipes outputs but not non-vanilla outputs.

Update 3: I forgot to mention this before but ores in specific are not the problem. They do in fact work if you're working with something like Coal ore. Since all it is outputing is coal which is a vanilla item it works fine. It also works with modded ores like Nether Coal ore. 

The problem seems to have less to do with the config files and more with the script files though more research is necessary to make a definitive conclusion.

**Comment by Pikachyuu** [comment]
> **Modpack version**
> 1.82a
> 
> **Issue**
> I know that this issue has been brought up before but I have some information to shed on the subject matter. So after coming across the bug myself while trying to turn coal into coal dust I decided to do a bit of experimenting. My first thought was to just start throwing a bunch of different items in and see if any of them would crush. The results of that is that there were certain items that would crush at the appropriate speed and others that wouldn't. After looking at the commonalities between the items I came to the conclusion that Items with a modded primary output take virtually no RF (I would say around 20RF per second) resulting in the long crushing time on the contrary items with a vanilla primary output take the appropriate amount of RF in the appropriate amount of time. There is an exception though. Crushed primary outputs from 'Immersive Engineering' itself seem to work perfectly fine. The input material does not seem to matter as I have tested it with both vanilla and non-vanilla items. I would assume the cause of this would have something to do with the link between the localized outputs and modded ones. Something which is causing the RF input amount into the machine to be much lower when dealing with non-localized outputs rather then with localized outputs. I would assume it's some sort of mod interference but there are other possibilities as well. I will continue with further testing but for now this is as much as I have figured out. If I have any other updates I will post them here.

**Comment by FreekMencke** [comment]
Duplicate of #1841

**Comment by Pikachyuu** [comment]
@FreekMencke 

> Duplicate of #1841

As I stated before I know this has already been brought up but I thought it would be better to shed some new light on it cause by what I've seen there isn't that much known about it or what causes it. I thought of writing all this information there but I was afraid it would get less attention on the matter so I decided to make it an entire separate issue. It's not entirely a duplicate it is just addressing the same problem. I wanted to bring forth a more detailed description of the problem to try and find a solution to it. I would appreciate it if it wasn't just written off as a duplicate of the original post.
